plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2907,7 +2908,8 @@ private static void validateTimestampColumns(List<ColumnMetadata> columns, HiveT
     private static void validateTimestampTypes(Type type, HiveTimestampPrecision precision)
     {
         if (type instanceof TimestampType) {
-            if (((TimestampType) type).getPrecision() != precision.getPrecision()) {
+            int typePrecision = ((TimestampType) type).getPrecision();
+            if (typePrecision >= MILLISECONDS.getPrecision() && typePrecision != precision.getPrecision()) {
```

Comment:
We have been debating this over here https://trinodb.slack.com/archives/CJ6UC075E/p1611149002013300?thread_ts=1611148282.012800&cid=CJ6UC075E (in context of Iceberg, but Hive would be similar) but apparently we did not reach conclusion quite yet.

Let's remove this change from this PR.

Reply:
I added it here since the default `timestamp`/`datetime` for Mysql is with precision `0` and without this change all `CTAS`s to hive from Mysql are failing without explicit casting from `timestamp(0`) to `timestamp(3)` 
I think it's safe to allow hive create `timestamp(3)` for cases that the source type is less than `3` (Actually it's sufficient to limit it only to precision of `0`)

Reply:
This is a preexisting problem, affecting every connector which supports parametric timestamps. Not really specific to MySQL.